mlflow/genai/judges/utils.py

```diff
@@ -26,32 +35,263 @@ def _sanitize_justification(justification: str) -> str:
     return justification.replace("Let's think step by step. ", "")
 
 
-@record_usage_event(InvokeCustomJudgeModelEvent)
-def invoke_judge_model(
-    model_uri: str, prompt: str, assessment_name: str, num_retries: int = 10
-) -> Feedback:
-    """
-    Invoke the judge model.
+@dataclass
+class InvokeDatabricksModelOutput:
+    response: str
+    request_id: str | None
+    num_prompt_tokens: int | None
+    num_completion_tokens: int | None
 
-    First, try to invoke the judge model via litellm. If litellm is not installed,
-    fallback to native parsing using the AI Gateway adapters.
+
+def _parse_databricks_model_response(
+    res_json: dict[str, Any], headers: dict[str, Any]
+) -> InvokeDatabricksModelOutput:
+    """
+    Parse and validate the response from a Databricks model invocation.
 
     Args:
-        model_uri: The model URI.
-        prompt: The prompt to evaluate.
-        assessment_name: The name of the assessment.
-        num_retries: Number of retries on transient failures when using litellm.
+        res_json: The JSON response from the model
+        headers: The response headers
+
+    Returns:
+        InvokeDatabricksModelOutput with parsed response data
+
+    Raises:
+        MlflowException: If the response structure is invalid
     """
+    # Validate and extract choices
+    choices = res_json.get("choices", [])
+    if not choices:
+        raise MlflowException(
+            "Invalid response from Databricks model: missing 'choices' field",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    first_choice = choices[0]
```

Comment:
```suggestion
    first_choice = choices[0]
    message = first_choice.get("first_choice")
```

let's extract message here. `first_choice["message"]` looks unsafe.

Reply:
We check right after this line whether "message" is in first_choice. We throw if it is not. Is that still unsafe?

Reply:
In general, `dictionary["key"]` is unsafe. I always try to minimize code that LOOKS unsafe (I don't care if it's really unsafe or not). 


```python
dictionary["key"]  # **LOOKS** unsafe, need to take a look at the line before this line. There might a check.

# vs.

dictionary.get("key")  # **IS** safe. Unsafe operations might follow but this line does LOOK safe.
```

Reply:
Gotcha, replaced content line with this: `content = first_choice.get("message", {}).get("content")`